users/engler/fairseq/training.py

```diff
@@ -91,15 +91,28 @@ def __init__(
         # Inputs:
         self.fairseq_hydra_config = fairseq_hydra_config
         self.command_line_args = command_line_args or []
-        stored_epochs = list(range(save_interval, max_epoch, save_interval)) + [
-            max_epoch
-        ]
+        self.save_interval = save_interval or fairseq_hydra_config.config_dict.get(
+            "checkpoint", {}
+        ).get("save_interval", None)
```

Comment:
is it possible to do something like!?

`self.save_interval = fairseq_hydra_config.config_dict.get("checkpoint", {}).get("save_interval", save_interval)`

then the default would be save_interval if the fairseq_hydra_config does not set the value. I think that would also save a bit of the logic in `_get_run_cmd`.

same for `max_epoch`

Reply:
But this would change the default behavior. Now, if `save_interval` is not `None` in the init, we use this and overwrite what is specified in the config. Your suggestion would be the other way around. I think it is more intuitive to take the input argument if both are not `None`.

Reply:
thats true. I think this is disambiguous.. mabye add a warning if both are not `None`?

Reply:
Yes, I did that now.